frontend/src/metabase/containers/AdHocQuestionLoader.jsx

```diff
@@ -0,0 +1,124 @@
+/* @flow */
+import React from "react";
+import { connect } from "react-redux";
+
+// things that will eventually load the quetsion
+import { deserializeCardFromUrl } from "metabase/lib/card";
+import { loadMetadataForCard } from "metabase/query_builder/actions";
+import { getMetadata } from "metabase/selectors/metadata";
+
+import Question from "metabase-lib/lib/Question";
+
+// type annotations
+import type Metadata from "metabase-lib/lib/metadata/Metadata";
+
+/*
+ * AdHocQuestionLoader
+ *
+ * Load a transient quetsion via its encoded URL and return it to the calling
+ * component
+ *
+ * @example
+ *
+ * Render prop style
+ * import AdHocQuestionLoader from 'metabase/containers/AdHocQuestionLoader'
+ *
+ * // assuming
+ * class ExampleAdHocQuestionFeature extends React.Component {
+ *    render () {
+ *      return (
+ *        <AdHocQuestionLoader questionId={this.props.params.questionId}>
+ *        { (question) => {
+ *
+ *        }}
+ *        </SavedQuestion>
+ *      )
+ *    }
+ * }
+ *
+ * @example
+ *
+ * The raw un-connected component is also exported so we can unit test it
+ * without the redux store.
+ */
+
+type Props = {
+  children: Function,
+  loadMetadataForCard: Function,
```

Comment:
These are ok but if you want to be more precise you can do `(card: Card) => Promise<void>`